mm2src/coins/eth.rs

```diff
@@ -3355,6 +3490,166 @@ fn rpc_event_handlers_for_eth_transport(ctx: &MmArc, ticker: String) -> Vec<RpcT
 #[inline]
 fn new_nonce_lock() -> Arc<AsyncMutex<()>> { Arc::new(AsyncMutex::new(())) }
 
+pub async fn eth_coin_from_conf_and_request_v2(
+    ctx: &MmArc,
+    ticker: &str,
+    conf: &Json,
+    req: EthActivationV2Request,
+    priv_key: &[u8],
+    protocol: CoinProtocol,
+) -> Result<EthCoin, MmError<EthActivationV2Error>> {
+    if req.nodes.is_empty() {
+        return Err(EthActivationV2Error::AtLeastOneNodeRequired(
+            "Enable request for ETH coin must have at least 1 node".to_string(),
+        )
+        .into());
+    }
+
+    let mut rng = small_rng();
+    let mut req = req;
+    req.nodes.as_mut_slice().shuffle(&mut rng);
+    drop_mutability!(req);
+
+    let mut nodes = vec![];
+    for node in req.nodes.iter() {
+        let uri = node
+            .url
+            .parse()
+            .map_err(|_| EthActivationV2Error::InvalidPayload(format!("{} could not be parsed.", node.url)))?;
+
+        nodes.push(Web3TransportNode {
+            uri,
+            gui_auth: node.gui_auth,
+        });
+    }
+    drop_mutability!(nodes);
+
+    if req.swap_contract_address == Address::default() {
+        return Err(
+            EthActivationV2Error::InvalidPayload("swap_contract_address can't be zero address".to_string()).into(),
+        );
+    }
+
+    if let Some(fallback) = req.fallback_swap_contract {
+        if fallback == Address::default() {
+            return Err(EthActivationV2Error::InvalidPayload(
```

Comment:
For clarity, a more specific error variant is preferred: `InvalidFallbackSwapContract` or maybe even `FallbackSwapContractIsZero`.